src/Unleash/ClientFactory/UnleashClientFactory.cs

```diff
@@ -25,11 +26,19 @@ public IUnleash CreateClient(UnleashSettings settings, bool synchronousInitializ
                 settings.ScheduleFeatureToggleFetchImmediatly = false;
                 settings.ThrowOnInitialFetchFail = true;
                 var unleash = new DefaultUnleash(settings, strategies);
-                TaskFactory
-                    .StartNew(() => unleash.services.FetchFeatureTogglesTask.ExecuteAsync(CancellationToken.None))
-                    .Unwrap()
-                    .GetAwaiter()
-                    .GetResult();
+                try
+                {
+                    TaskFactory
+                        .StartNew(() => unleash.services.FetchFeatureTogglesTask.ExecuteAsync(CancellationToken.None))
+                        .Unwrap()
+                        .GetAwaiter()
+                        .GetResult();
+                }
+                catch (Exception ex)
+                {
+                    unleash.Dispose();
+                    throw ex;
```

Comment:
I think Github has a fair point here. We probably want this
```suggestion
                    throw;
```